entrypoint.py

```diff
@@ -43,14 +59,114 @@ def check_max_answer_allowed(self, series, json_response_df):
         return False
 
     def validate_uploaded_data(self, series, json_response_df):
-        return self.check_max_answer_allowed(series, json_response_df)
+        data_type_verification = self.check_data_type(series, json_response_df)
+        max_answer_verification = self.check_max_answer_allowed(series, json_response_df)
+        return data_type_verification or max_answer_verification
+
+    def split_and_clean(self, curr_list_answers, token=r'[,=]'):
+        curr_list_answers_list = list(
+            curr_list_answers.str.split(token).values)
+        curr_list_answers_list = list(
+            map(str.strip, [item for sublist in curr_list_answers_list for item in sublist]))
+        return curr_list_answers_list
+
+    def check_date(self, series):
+        for value in series.values:
+            try:
+                datetime.date.fromisoformat(value)
+            except ValueError:
+                return True
+        return False
+
+    def check_time(self, series):
+        for value in series.values:
+            try:
+                datetime.time.fromisoformat(value)
+            except ValueError:
+                return True
+        return False
+
+    def check_datetime(self, series):
+        for value in series.values:
+            try:
+                datetime.datetime.fromisoformat(value)
+            except ValueError:
+                return True
+        return False
+
+    def check_categorical(self, series, measure_level):
+        curr_list_answers_list = self.split_and_clean(measure_level, r'[,=]')
+        for combined_value in series.values:
+            # Split the combined value based on ';'
+            if not isinstance(combined_value, str):
+                combined_value = str(combined_value)
+            values = combined_value.split(';')
+            for value in values:
+                if value not in curr_list_answers_list:
+                    return True
+        return False
+
+    def check_ordinal(self, series, measure_level):
+        curr_list_answers_list = self.split_and_clean(measure_level, r'[,]')
+        for value in series.values:
+            if value not in curr_list_answers_list:
+                return True
+        return False
+
+    def check_numeric(self, series):
+        for value in series.values:
+            try:
+                float(value)
+            except ValueError:
+                return True
+        return False
+
+    def check_boolean(self, series):
+        accepted_boolean_values = ["Yes", "Y", "No", "N"]
+        for value in series.values:
+            if value not in accepted_boolean_values:
```

Comment:
It should be also considered that the values can be all with lower case, or upper case. Lower case or upper case shouldn't affect the check.